File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let min = prices[0];
    let profit = 0;
    for(let i =1; i<prices.length; i++){
        if(min > prices[i]){
            min = prices[i];  
        }
        if(profit < (prices[i]-min)){
            profit = (prices[i]-min)
        }   
    }
    return profit
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min = prices[0]
        profit = 0
        for i in range(len(prices)):
            if (profit < (prices[i]-min)):
                profit = prices[i]-min
            if min > prices[i]:
                min = prices[i]
        return profit